shared/src/commonTest/kotlin/co/nimblehq/avishek/phong/kmmic/domain.usecase/LogInUseCaseTest.kt

```diff
@@ -0,0 +1,74 @@
+package co.nimblehq.avishek.phong.kmmic.domain.usecase
+
+import co.nimblehq.avishek.phong.kmmic.domain.model.Token
+import co.nimblehq.avishek.phong.kmmic.domain.repository.AuthenticationRepository
+import io.kotest.matchers.shouldBe
+import io.mockative.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.flow.collect
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+
+@ExperimentalCoroutinesApi
+class LogInUseCaseTest {
+
+    @Mock
+    private val mockRepository = mock(classOf<AuthenticationRepository>())
+
+    private lateinit var useCase: LogInUseCase
+
+    private val mockThrowable = Throwable("mock")
+    private val mockToken = Token(
+        "",
+        "",
+        0,
+        "",
+        0
+    )
```

Comment:
I think this should work too making the setup simpler.
```suggestion
    private val mockToken = mock(classOf<Token>())
```

Reply:
It doesn't work in this case. I got an error while trying to do that.

![Screenshot 2023-05-08 at 14 38 19](https://user-images.githubusercontent.com/22606906/236764585-a66f03a2-fe4c-4dc5-9f80-0b991bcf12d1.png)